### PR TITLE
clang: Upgrade to 19.x release

### PR DESCRIPTION
**Pull request by kraj**
---
### Contributor checklist

- [x] Changes have been tested
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

### CODE REVIEW

File: dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch [hunk 1]
```diff
@@ -1,27 +1,24 @@
 From 16186113346c268a0bb45424ba1c41768b7e94cf Mon Sep 17 00:00:00 2001
 From: Khem Raj <raj.khem@gmail.com>
 Date: Fri, 16 Feb 2024 10:14:41 -0800
-Subject: [PATCH 4/4] cmake: Bump max LLVM version to 18+
+Subject: [PATCH] cmake: Bump max LLVM version to 19
 
-Upstream-Status: Submitted [https://github.com/bpftrace/bpftrace/pull/3009]
+Upstream-Status: Pending
```
Comment (shr-project) [inline dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch:6]: Was it un-submitted or just changed Upstream-Status by accident?

File: dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace_0.21.2.bb [hunk 2]
```diff
@@ -19,14 +19,10 @@ RDEPENDS:${PN} += "bash python3 xz"
...
 
 SRC_URI = "git://github.com/iovisor/bpftrace;branch=master;protocol=https \
-           file://0001-replace-python-with-python3-in-the-test.patch \
-           file://0002-ast-Repace-getInt8PtrTy-with-getPtrTy.patch \
-           file://0003-ast-Adjust-to-enum-changes-in-llvm-18.patch \
-           file://0004-cmake-Bump-max-LLVM-version-to-18.patch \
-           file://0001-use-64bit-alignment-for-map-counter-atomic-add.patch \
            file://run-ptest \
+           file://0001-cmake-Bump-max-LLVM-version-to-19.patch \
 "
```
Comment (shr-project) [inline dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace_0.21.2.bb:24]: Do you happen to have buildpaths disabled or lowered to WARN_QA?

With ptest enabled I'm seeing (with older 0.20.1 version):
http://errors.yoctoproject.org/Errors/Build/190235/
```
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/lib/bpftrace/ptest/tests/bpftrace_test in package bpftrace-ptest contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/lib/bpftrace/ptest/tests/testprogs/cmake_install.cmake in package bpftrace-ptest contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/lib/bpftrace/ptest/tests/testprogs/CTestTestfile.cmake in package bpftrace-ptest contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/lib/bpftrace/ptest/tests/testlibs/cmake_install.cmake in package bpftrace-ptest contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/lib/bpftrace/ptest/tests/testlibs/CTestTestfile.cmake in package bpftrace-ptest contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/src/debug/bpftrace/0.20.1+git/parser.tab.hh in package bpftrace-src contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/src/debug/bpftrace/0.20.1+git/lex.yy.cc in package bpftrace-src contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/src/debug/bpftrace/0.20.1+git/parser.tab.cc in package bpftrace-src contains reference to TMPDIR [buildpaths]
ERROR: bpftrace-0.20.1+git-r0 do_package_qa: QA Issue: File /usr/src/debug/bpftrace/0.20.1+git/location.hh in package bpftrace-src contains reference to TMPDIR [buildpaths]
```

File: dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch (hunk 1 above)
Comment (kraj) [reply dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch:6]: the original patch was for clang18 which was merged. This is for clang19 which is not yet submitted.

File: dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace_0.21.2.bb (hunk 2 above)
Comment (kraj) [reply dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace_0.21.2.bb:24]: yes in yoe `buildpaths` is downgraded to be a warning.

File: dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch (hunk 1 above)
Comment (kraj) [reply dynamic-layers/openembedded-layer/recipes-devtools/bpftrace/bpftrace/0001-cmake-Bump-max-LLVM-version-to-19.patch]: I have submitted it upstream and updated the patch

**Comment by shr-project** [comment]
FYI: I've noticed one more small issue introduced by this PR in some multilib builds:
```
ERROR: QA Issue: lib32-clang: Files/directories were installed but not shipped in any package:
  /usr/lib/libscanbuild
  /usr/lib/libscanbuild/intercept.py
  /usr/lib/libscanbuild/analyze.py
  /usr/lib/libscanbuild/clang.py
  /usr/lib/libscanbuild/report.py
  /usr/lib/libscanbuild/arguments.py
  /usr/lib/libscanbuild/shell.py
  /usr/lib/libscanbuild/compilation.py
  /usr/lib/libscanbuild/__init__.py
  /usr/lib/libscanbuild/resources
  /usr/lib/libscanbuild/resources/sorttable.js
  /usr/lib/libscanbuild/resources/selectable.js
  /usr/lib/libscanbuild/resources/scanview.css
Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
lib32-clang: 13 installed and not shipped files. [installed-vs-shipped]
```

In these builds we use a bit strange libdir /usr/lib32 (instead of usr/lib and usr/lib64), but other parts of clang respect that, only libscanbuild doesn't. Will investigate and send fix (I don't think it should block this PR, it's just FYI).

**Comment by shr-project** [comment]
installed-vs-shipped QA issue fixed in:
https://github.com/shr-project/meta-clang/commit/ff430da341c0489543c9f806a0ff54a795b7c80f

if you want to pick it to this PR (or I can open separate PR on top of this)

**Comment by kraj** [comment]
> installed-vs-shipped QA issue fixed in: [shr-project@ff430da](https://github.com/shr-project/meta-clang/commit/ff430da341c0489543c9f806a0ff54a795b7c80f)
> 
> if you want to pick it to this PR (or I can open separate PR on top of this)

yeah I have cherry-picked it into clang19 PR and also to my llvm fork.

**Pull request merged**